src/core/layout/qgscompositionconverter.cpp

```diff
@@ -0,0 +1,1414 @@
+  {
+    layoutItem->setSymbol( QgsSymbolLayerUtils::loadSymbol<QgsFillSymbol>( shapeStyleSymbolElem, context ) );
+  } */
+  else
+  {
+    //upgrade project file from 2.0 to use symbol styling
+    QgsStringMap properties;
+    properties.insert( QStringLiteral( "color" ), QgsSymbolLayerUtils::encodeColor( layoutItem->brush().color() ) );
+    if ( layoutItem->hasBackground() )
+    {
+      properties.insert( QStringLiteral( "style" ), QStringLiteral( "solid" ) );
+    }
+    else
+    {
+      properties.insert( QStringLiteral( "style" ), QStringLiteral( "no" ) );
+    }
+    if ( layoutItem->hasFrame() )
+    {
+      properties.insert( QStringLiteral( "style_border" ), QStringLiteral( "solid" ) );
+    }
+    else
+    {
+      properties.insert( QStringLiteral( "style_border" ), QStringLiteral( "no" ) );
+    }
+    properties.insert( QStringLiteral( "color_border" ), QgsSymbolLayerUtils::encodeColor( layoutItem->pen().color() ) );
+    properties.insert( QStringLiteral( "width_border" ), QString::number( layoutItem->pen().widthF() ) );
+
+    //for pre 2.0 projects, shape color and outline were specified in a different element...
+    QDomNodeList outlineColorList = itemElem.elementsByTagName( QStringLiteral( "OutlineColor" ) );
+    if ( !outlineColorList.isEmpty() )
+    {
+      QDomElement frameColorElem = outlineColorList.at( 0 ).toElement();
+      bool redOk, greenOk, blueOk, alphaOk, widthOk;
+      int penRed, penGreen, penBlue, penAlpha;
+      double penWidth;
+
+      penWidth = itemElem.attribute( QStringLiteral( "outlineWidth" ) ).toDouble( &widthOk );
+      penRed = frameColorElem.attribute( QStringLiteral( "red" ) ).toDouble( &redOk );
+      penGreen = frameColorElem.attribute( QStringLiteral( "green" ) ).toDouble( &greenOk );
+      penBlue = frameColorElem.attribute( QStringLiteral( "blue" ) ).toDouble( &blueOk );
+      penAlpha = frameColorElem.attribute( QStringLiteral( "alpha" ) ).toDouble( &alphaOk );
+
+      if ( redOk && greenOk && blueOk && alphaOk && widthOk )
+      {
+        properties.insert( QStringLiteral( "color_border" ), QgsSymbolLayerUtils::encodeColor( QColor( penRed, penGreen, penBlue, penAlpha ) ) );
+        properties.insert( QStringLiteral( "width_border" ), QString::number( penWidth ) );
+      }
+    }
+    QDomNodeList fillColorList = itemElem.elementsByTagName( QStringLiteral( "FillColor" ) );
+    if ( !fillColorList.isEmpty() )
+    {
+      QDomElement fillColorElem = fillColorList.at( 0 ).toElement();
+      bool redOk, greenOk, blueOk, alphaOk;
+      int fillRed, fillGreen, fillBlue, fillAlpha;
+
+      fillRed = fillColorElem.attribute( QStringLiteral( "red" ) ).toDouble( &redOk );
+      fillGreen = fillColorElem.attribute( QStringLiteral( "green" ) ).toDouble( &greenOk );
+      fillBlue = fillColorElem.attribute( QStringLiteral( "blue" ) ).toDouble( &blueOk );
+      fillAlpha = fillColorElem.attribute( QStringLiteral( "alpha" ) ).toDouble( &alphaOk );
```

Comment:
Is there a reason to use the `toDouble()` method for integer variables like `fillRed`, `fillGreen`, ...?